code/game/objects/effects/aliens.dm

```diff
@@ -196,21 +196,15 @@
 		qdel(src)
 		return
 
-	direction_loop:
-		for(var/dirn in cardinal)
-			var/turf/T = get_step(src, dirn)
+	for(var/turf/T in U.GetAtmosAdjacentTurfs())
 
-			if (!istype(T) || T.density || locate(/obj/structure/alien/weeds) in T || istype(T, /turf/space))
-				continue
+		if (locate(/obj/structure/alien/weeds) in T || istype(T, /turf/space))
+			continue
 
-			if(!linked_node || get_dist(linked_node, src) > linked_node.node_range)
-				return
-
-			for(var/obj/O in T)
-				if(O.density)
-					continue direction_loop
+		if(!linked_node || get_dist(linked_node, src) > linked_node.node_range)
+			return

```

Comment:
Minor complaint: This doesn't change for the turf in the loop, thus it shouldn't be in the for loop, it should be before it.

Would save running get_dist 3 times,